hpx/traits/is_bitwise_serializable.hpp

```diff
@@ -0,0 +1,28 @@
+//  Copyright (c) 2014 Thomas Heller
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef HPX_TRAITS_IS_BITWISE_SERIALIZABLE_HPP
+#define HPX_TRAITS_IS_BITWISE_SERIALIZABLE_HPP
+
+#include <boost/mpl/bool.hpp>
+#include <boost/type_traits/is_arithmetic.hpp>
+
+namespace hpx { namespace traits {
+    template <typename T>
+    struct is_bitwise_serializable
+      : boost::is_arithmetic<T>
```

Comment:
Mismatching endianess between two communicating peers should be detected while constructing the input/output archive and should disable array optimizations globally, So I think that might not be a real problem.